distributed/core.py

```diff
@@ -519,7 +519,7 @@ async def handle_comm(self, comm):
                             result = asyncio.ensure_future(result)
                             self._ongoing_coroutines.add(result)
                             result = await result
-                    except (CommClosedError, CancelledError):
+                    except (CommClosedError, CancelledError, asyncio.CancelledError):
```

Comment:
I'm not sure about this. What determines which exception is raised (CancelledError is concurrent.futures.CancelledError, which is raised by tornado)? The event loop being used, or the choice of ``@gen_coroutine`` vs. ``async def``?

Anyway I didn't find any failure here in real life - this is just defensive programming.

Reply:
I'd actually never expect a `concurrent.futures.CancelledError` here, but would expect a `asyncio.CancelledError` to be possible. If a task is ever cancelled while waiting on `handle_comm`, a `asyncio.CancelledError` should be raised. If we're seeing both types of cancelled errors here, we should probably figure out how to convert to only one (I'd prefere the asyncio version, which should be what's hit most places in the code base).

Reply:
Would you like me to change it to

```python

                    except (CommClosedError, asyncio.CancelledError):
                        ...
                    except CancelledError as e:  # pragma: nocover
                        raise AssertionError("Unexpected exception") from e
```

and see if the path is ever hit? Or should I just clean it away?

Reply:
I think it'd be good to remove the catch for the concurrent futures version, if for no other reason than readability. I'd probably delete it and see if any tests fail, but up to you.

Reply:
done